packages/core/saved-objects/core-saved-objects-base-server-internal/src/migration/document_migrator.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { SavedObjectUnsanitizedDoc } from '@kbn/core-saved-objects-server';
+
+/**
+ * Manages transformations of individual documents.
+ */
+export interface IDocumentMigrator {
+  /**
+   * Migrates a document to its latest version.
+   */
+  migrate(
+    doc: SavedObjectUnsanitizedDoc,
+    options?: DocumentMigrateOptions
+  ): SavedObjectUnsanitizedDoc;
+
```

Comment:
I extracted the document migrator interface (and took the opportunity to rename it from the old `VersionedTransformer` name), as it now needs to live at the same place as the Kibana migrator interface (given the kibana migrator now has a signature referencing the doc migrator).

But again, it was just tech debt, those interfaces should have been grouped from the beginning.